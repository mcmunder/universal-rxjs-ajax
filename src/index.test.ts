import {Observable} from 'rxjs'
import {map} from 'rxjs/operators'
import {request} from './index'

type Repo = {
  name: string
}

describe('universal-rx-js-ajax', () => {
  test('request returns Observable', () => {
    const result = request({url: 'http://some-api.af'})
    expect(result instanceof Observable).toBeTruthy()
    expect(result).toMatchSnapshot()
  })

  test('README example works', () => {
    // config as accepted by Observable.ajax()
    const config = {
      url: 'https://api.github.com/orgs/Reactive-Extensions/repos',
      method: 'GET', // and so on...
    }

    request<Repo[]>(config)
      .pipe(map(({response}) => response.map((repo: Repo) => repo.name)))
      .subscribe((repoNames) => {
        expect(repoNames.includes('RxJS'))
      })
  })
})
