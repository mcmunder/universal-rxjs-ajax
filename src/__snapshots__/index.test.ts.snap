// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`universal-rx-js-ajax request returns Observable 1`] = `
Observable {
  "_subscribe": [Function],
}
`;
